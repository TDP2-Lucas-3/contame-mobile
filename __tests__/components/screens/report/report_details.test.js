import React from 'react';
import {shallow} from 'enzyme';
import ReportDetails from '../../../../src/components/screens/reports/report_details';


describe('ReportDetails', () => {
  let render;
  let props;

  beforeEach(() => {
    props = {
      title: 'a report',
      description: 'some description',
    };
    render = (appProps) => shallow(<ReportDetails {...appProps} />);
  });

  it('renders without crashing', () => {
    expect(() => render(props)).not.toThrowError();
  });

  it('renders all values correctly', () => {
    expect(render(props)).toMatchSnapshot();
  });
});
