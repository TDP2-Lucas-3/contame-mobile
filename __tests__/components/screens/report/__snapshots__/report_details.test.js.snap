// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDetails renders all values correctly 1`] = `
<Themed.Card
  containerStyle={
    Object {
      "borderRadius": 10,
      "elevation": 10,
      "padding": 0,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
    }
  }
>
  <Themed.CardImage
    source={
      Object {
        "uri": "some image data",
      }
    }
    style={
      Object {
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
      }
    }
    testID="card_image"
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 10,
      }
    }
  >
    <View>
      <Themed.CardTitle
        style={
          Object {
            "alignSelf": "flex-start",
          }
        }
      >
        a report
      </Themed.CardTitle>
      <Themed.Text
        style={
          Object {
            "color": "grey",
          }
        }
      >
        a few seconds ago
      </Themed.Text>
    </View>
    <View>
      <Themed.Icon
        name="chevron-right"
        size={30}
      />
    </View>
  </View>
</Themed.Card>
`;
